src/main/java/com/google/sps/servlets/CreateRoomServlet.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.sps.daos.RoomDao;
+
+@WebServlet("/newRoom")
+public class CreateRoomServlet extends HttpServlet {
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    // Generate room key
+    RoomDao dao = (RoomDao) this.getServletContext().getAttribute("roomDao");
+    String roomId = dao.createNewRoom("5GoXkueDimFjk1wEBLhV");
+    
+    JsonObject jsonObj = new JsonObject();
+    jsonObj.addProperty("roomId", roomId);
+    
+    String json = new Gson().toJson(jsonObj);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Generate room key
+    RoomDao dao = (RoomDao) this.getServletContext().getAttribute("roomDao");
+    dao.createNewRoom("5GoXkueDimFjk1wEBLhV");
```

Comment:
Isn't `RoomDao` an interface?
```suggestion
    DatabaseRoomDao dao = (RoomDao) this.getServletContext().getAttribute("roomDao");
    dao.createNewRoom("5GoXkueDimFjk1wEBLhV");
```

Reply:
Yes it is, but @brettapeters told us that this is the best way to have the DAO.

Reply:
Right. Same as you would write `List<String> names = new ArrayList<>();` `List` is an interface and `ArrayList` is an implementation of a `List`. This servlet doesn't need to know which implementation of RoomDao it's using.

Reply:
Kind of sort of related to the "I" in SOLID. https://en.wikipedia.org/wiki/SOLID